File: Desktop/3a/sem2/api/projet-sante-connectee-Event/src/Entity/Produit.php
<?php

namespace App\Entity;


use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
class Produit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: "integer")]
    private $id;

    #[ORM\Column(type: "string", length: 255)]
    private $nom;

    #[ORM\Column(type: "text", nullable: true)]
    private $description;

    #[ORM\Column(type: "decimal", scale: 2)]
    private $prix;

    #[ORM\Column(type: "integer")]
    private $quantite;

    // Définir la relation ManyToOne avec l'entité Categorie
    #[ORM\ManyToOne(targetEntity: Categorie::class)]
    #[ORM\JoinColumn(name: "categorie_id", referencedColumnName: "id")]
    private $categorie;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $image_path = null;  // Champ imagePath ajouté (anciennement 'image')

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;
        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;
        return $this;
    }

    // Getter et Setter pour la propriété 'categorie'
    public function getCategorie(): ?Categorie
    {
        return $this->categorie;
    }

    public function setCategorie(?Categorie $categorie): self
    {
        $this->categorie = $categorie;
        return $this;
    }

    // Getter pour récupérer le chemin de l'image
    public function getimage_path(): ?string
    {
        return $this->image_path;
    }

    // Setter pour définir le chemin de l'image
    public function setimage_path(?string $imagePath): self
    {
        $this->image_path = $imagePath;
        return $this;
    }

}

File: src/main/java/models/Produit.java
package models;

public class Produit {

    int id;
    String nom;
    String description;
    double prix;
    int quantite;
    int categorie_id;
    String image_path;
    String model3D;

    public Produit() {}

    public Produit(int id, String nom, String description, double prix, int quantite, int categorie_id, String image_path, String model3D) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.quantite = quantite;
        this.categorie_id = categorie_id;
        this.image_path = image_path;
        this.model3D = model3D;
    }

    public Produit(String nom, String description, double prix, int quantite, int categorie_id, String image_path, String model3D) {
        this.nom = nom;
        this.description = description;
        this.prix = prix;
        this.quantite = quantite;
        this.categorie_id = categorie_id;
        this.image_path = image_path;
        this.model3D = model3D;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public int getCategorie() {
        return categorie_id;
    }

    public void setCategorie(int categorieId) {
        this.categorie_id = categorieId;
    }

    public String getimage_path() {
        return image_path;
    }

    public void setimage_path(String image_path) {
        this.image_path = image_path;
    }

    public String getModel3D() {
        return model3D;
    }

    public void setModel3D(String model3D) {
        this.model3D = model3D;
    }

    @Override
    public String toString() {
        return "Produit{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", description='" + description + '\'' +
                ", prix=" + prix +
                ", quantite=" + quantite +
                ", categorieId=" + categorie_id +
                ", image_path='" + image_path + '\'' +
                ", model3D='" + model3D + '\'' +
                '}';
    }
}